lib/src/recognitionService/kbWhisperLocal.ts

```diff
@@ -0,0 +1,129 @@
+import type { SpeechProbabilities } from "@ricky0123/vad-web/dist/models";
+import {
+  type RecognitionService,
+  type RecognitionServiceListenOptions
+} from './interface';
+
+import { MicVAD, utils } from '@ricky0123/vad-web';
+import { pipeline } from '@xenova/transformers';
+
+export interface kbWhisperLocalOptions extends RecognitionServiceListenOptions {
+  lang?: string;
+}
+
+export class kbWhisperlocal implements RecognitionService {
+  private options?: kbWhisperLocalOptions;
+  private vad?: Awaited<ReturnType<typeof MicVAD.new>>;
+  private transcriber?: any;
+
+  private listeningState: "listening" | "inactive" = "inactive";
+  private inputSpeechState: "speaking" | "idle" = "idle";
+
+  constructor(options?: kbWhisperLocalOptions) {
+    this.options = options;
+  }
+
+  private async loadTranscriber() {
+    if (!this.transcriber) {
+      this.transcriber = await pipeline(
+        'automatic-speech-recognition',
+        'KBLab/kb-whisper-small',
+        {
+          local_files_only: true
+        }
+      );
+    }
+  }
```

Comment:
I'd recommend to always have a function do what it says. The function is named loadTranscriber, which implives it **will** load a transcriber. What the function actually does is to maybe load a transcriber, if there isn't one available. I would suggest to either rename the function, or move the the "is defined" logic outside the function.
```suggestion
  private async loadTranscriber() {
    this.transcriber = await pipeline(
      'automatic-speech-recognition',
      'KBLab/kb-whisper-small',
      {
        local_files_only: true
      }
    );
  }
```